source/simulator/helper_functions.cc

```diff
@@ -1529,13 +1529,39 @@ namespace aspect
 
 
 
+  template <int dim>
+  void Simulator<dim>::update_solution_vectors_with_reaction_results (const unsigned int block_index,
+                                                                      const LinearAlgebra::BlockVector &distributed_vector,
+                                                                      const LinearAlgebra::BlockVector &distributed_reaction_vector)
+  {
+    solution.block(block_index) = distributed_vector.block(block_index);
+
+    // we have to update the old solution with our reaction update too
+    // so that the advection scheme will have the correct time stepping in the next step
+    LinearAlgebra::BlockVector tmp(distributed_vector);
```

Comment:
Could you do
```suggestion
    LinearAlgebra::BlockVector tmp(distributed_vector, false);
```
to avoid the actual copying of vector elements (which you don't need)?

Reply:
There is no copy constructor like that, but I used `reinit (vector, false)` instead, which does not copy entries.